MmSupervisorPkg/Plugin/PostBuildScanMmStandalone.py

```diff
@@ -0,0 +1,166 @@
+        """
+        env = (thebuilder.env.GetAllBuildKeyValues() |
+               thebuilder.env.GetAllNonBuildKeyValues())
+        build_report = Path(env.get("BUILDREPORT_FILE", ""))
+        if not build_report.exists():
+            logging.warning(
+                "Build report not present, skipping build report table "
+                "parsing."
+            )
+            return
+        self.dumpbin_path = self.find_dumpbin()
+
+        report = BuildReport(
+            build_report,
+            thebuilder.edk2path.WorkspacePath,
+            ",".join(thebuilder.edk2path.PackagePathList),
+            {}
+        )
+        report.BasicParse()
+
+        module_list = []
+        for module in report.Modules.values():
+            if module.FvName and module.Type == "MM_STANDALONE":
+                module_list.append(module.Name)
+        for efi_file in module_list:
+            workspace_path = thebuilder.edk2path.WorkspacePath
+            file_pattern = Path(workspace_path, "**", efi_file + ".efi")
+            path_to_efi_file = glob.glob(
+                file_pattern.as_posix(), recursive=True
+            )
+            if path_to_efi_file:
+                disassembly = self.disassemble_with_dumpbin(
+                    path_to_efi_file[0]
+                )
+                msr_hits = self.find_msr_instructions_with_context(
+                    disassembly
+                )
+                if msr_hits:
+                    logging.info(f"\nScanning: {efi_file}")
+                    for (formatted_line,) in msr_hits:
+                        logging.info(formatted_line)
+        return 0
+
+    def find_dumpbin(self) -> str:
+        """Find the path to the dumpbin.exe tool.
+
+        Returns:
+            str: Path to dumpbin.exe.
+
+        Raises:
+            EnvironmentError: If VsWhere executable cannot be located.
+            RuntimeError: If VsWhere execution fails.
+        """
+        vs_where_path = GetVsWherePath()
+        if vs_where_path is None:
+            raise EnvironmentError("Unable to locate the VsWhere Executable.")
+
+        cmd = (
+            "-latest -products * -requires "
+            "Microsoft.VisualStudio.Component.VC.Tools.x86.x64 "
+            "-find **\\dumpbin.exe"
+        )
+        a = StringIO()
+        ret = RunCmd(vs_where_path, cmd, outstream=a)
+        if ret != 0:
+            a.close()
+            raise RuntimeError(
+                f"Unknown Error while executing VsWhere: errcode {ret}."
+            )
+
+        possible_bins = a.getvalue().replace("\r\n", ";").split(";")
+        a.close()
+        return possible_bins[0]
+
+    def disassemble_with_dumpbin(self, file_path: str) -> List[str]:
+        """Disassemble a file using dumpbin.exe.
+
+        Args:
+            file_path: Path to the file to disassemble.
+
+        Returns:
+            List of disassembly lines.
+        """
+        try:
+            result = subprocess.run(

```

Comment:
since you are using pytools please use RunCmd (https://github.com/tianocore/edk2-pytool-library/blob/23fb161210598927e315f60a03215397e5236e06/edk2toollib/utility_functions.py#L144)

Reply:
fixed in latest